airflow/providers/google/cloud/transfers/gcs_to_bigquery.py

```diff
@@ -300,6 +305,16 @@ def execute(self, context: Context):
             impersonation_chain=self.impersonation_chain,
         )
         self.hook = hook
+        self.source_format = self.source_format.upper()
+        allowed_formats = [
```

Comment:
Why not declare this and validate `source_format` in the `__init__` to fail at dag parsing time instead of during the execution?

Reply:
Yes, looks like it will be better. I have added all appropriate changes.
Thanks!